magento/magento2
# [2.3] Database Media Storage - Design Config Save functions to be Database Media Storage aware

### Description (*)
The functions that process the uploads of the Favicon, Header Image and Transactional Email Logo files do not process the file correctly when in Database Media Storage Mode. This PR fixes that.

### Fixed Issues (if relevant)
1. magento/magento2#21672: Database Media Storage - Design Config fails to save transactional email logo correctly

### Manual testing scenarios (*)
1. Setup Magento
2. Change media storage mode to database
3. Synchronize
4. Save Settings
5. Content->Configuration->Edit->Transactional Emails
6. Upload New Image (but don't save)
7. Verify temporary email logo exists in pub/media
8. Verify temporary email logo exists in database
9. Save Config
10. Verify email logo exists in pub/media
11. Verify email logo exists in database

Repeat for favicon and header logo, ensuring that after save, the file exists correctly in the database.

### Contribution checklist (*)
